Makefile

```diff
@@ -58,40 +69,49 @@ biniou.cmxa: $(SOURCES) Makefile
 		-o biniou.cmxa -package "$(PACKS)" $(SOURCES)
 
 biniou.cmxs: biniou.cmxa
-	ocamlfind ocamlopt -shared -linkall -I . -o biniou.cmxs biniou.cmxa
+	ocamlfind ocamlopt -shared -linkall -I . -o $@ $^
 
-bdump$(EXE): $(SOURCES) bdump.ml
-	ocamlfind ocamlopt -o bdump$(EXE) $(FLAGS) \
-		-package $(PACKS) -linkpkg \
-		biniou.cmxa bdump.ml
+bdump.byte: biniou.cma bdump.ml
+	ocamlfind ocamlc -o $@ $(FLAGS) \
+		-package $(PACKS) -linkpkg $^
 
-test_biniou$(EXE): biniou.cmxa test_biniou.ml
-	ocamlfind ocamlopt -o test_biniou$(EXE) $(FLAGS) \
-		-package "$(PACKS) unix" -linkpkg \
-		biniou.cmxa test_biniou.ml
+bdump.native: biniou.cmxa bdump.ml
+	ocamlfind ocamlopt -o $@ $(FLAGS) \
+		-package $(PACKS) -linkpkg $^
+
+test_biniou.byte: biniou.cma test_biniou.ml
+	ocamlfind ocamlc -o $@ $(FLAGS) \
+		-package "$(PACKS) unix" -linkpkg $^
+
+test_biniou.native: biniou.cmxa test_biniou.ml
+	ocamlfind ocamlopt -o $@ $(FLAGS) \
+		-package "$(PACKS) unix" -linkpkg $^
 
 doc: doc/index.html
 doc/index.html: $(MLI)
 	mkdir -p doc
 	ocamlfind ocamldoc -d doc -html -package easy-format $(MLI)
 
-install: META
-	test ! -f bdump || cp bdump $(BINDIR)/
-	test ! -f bdump.exe || cp bdump.exe $(BINDIR)/
+install: META byte
+	if [ -f bdump.native ]; then \
+		cp bdump.native $(BINDIR)/bdump$(EXE); \
+	else \
+		cp bdump.byte $(BINDIR)/bdump$(EXE); \
+	fi
 	ocamlfind install biniou META \
-          $$(ls $(MLI) $(CMI) $(CMO) $(CMX) $(O) \
-             biniou.cma biniou.cmxa biniou.cmxs biniou.a)
+	  $(MLI) $(CMI) $(CMT) $(ANNOT) $(CMO) biniou.cma \
+	  -optional $(CMX) $(O) biniou.cmxa biniou.a biniou.cmxs
 
 uninstall:
-	test ! -f $(BINDIR)/bdump || rm $(BINDIR)/bdump
-	test ! -f $(BINDIR)/bdump.exe || rm $(BINDIR)/bdump.exe
+	test -f $(BINDIR)/bdump && rm $(BINDIR)/bdump
+	test -f $(BINDIR)/bdump.exe && rm $(BINDIR)/bdump.exe
```

Comment:
Does this change work for you? The `&&` tests are more natural but result in an error status, preventing make to continue. In order to ignore errors due to missing files, the recipes should start with `-` as explained in https://www.gnu.org/software/make/manual/html_node/Errors.html